### PR TITLE
[MRG, BUG] Make params private

### PR DESCRIPTION
**Pull request by alexrockhill**
Addresses https://github.com/jonescompneurolab/hnn-core/issues/307.

### CODE REVIEW

File: hnn_core/dipole.py [hunk 1]
```diff
@@ -65,8 +65,8 @@ def simulate_dipole(net, n_trials=None, record_vsoma=False,
         The Network object specifying how cells are
         connected.
     n_trials : int | None
-        The number of trials to simulate. If None, the value in
-        net.params['N_trials'] is used (must be >0)
+        The number of trials to simulate. If None, the 'N_trials' value
+        of the ``net`` params is used (must be >0)
```
Comment (jasmainak) [inline hnn_core/dipole.py:69]:
```suggestion
        of the params used to create ``net`` is used (must be >0)
```

**Comment by jasmainak** [comment]
looks good. Please set it to merge when ready. I think you need a more difficult issue now ;-)

@ntolley any comments?

**Comment by ntolley** [comment]
Edits look good so far! There's a few examples that print out information from `net.params` that need updating though. This one for instance: https://github.com/jonescompneurolab/hnn-core/blob/d4ba506cc37108701a244378c88222cfa9f0807c/examples/plot_simulate_evoked.py#L40

**Comment by ntolley** [comment]
@alexrockhill the linux build passed but macOS failed (it's a bit finicky... I really wish I knew why). Do you have the right permissions to restart the travis tests?

**Comment by jasmainak** [comment]
I restarted a bit. We'll wait a bit before giving @alexrockhill more superpowers ;-) Some nice PRs and you're on your way to HNN-superpowers

**Comment by cjayb** [comment]
Excellent!

**Pull request merged**